### PR TITLE
feat(1922): Add openPr method. BREAKING CHANGE: New method is required

### PR DESCRIPTION
**Pull request by klu909**
## Context

Sd onboarding feature requires an api to open pull request 

## Objective

Add a method `openPr` for opening pull request to support sd onboarding feature

## References

https://github.com/screwdriver-cd/screwdriver/issues/1922

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -560,6 +560,26 @@ class ScmBase {
     _getBranchList() {
         return Promise.reject(new Error('Not implemented'));
     }
+
+    /**
+     * Open a pull request on the repository with given file change
+     *
+     * @method addWebhook
```
Comment (tkyi) [inline index.js:567]:
```suggestion
     * @method openPr
```

File: index.js [hunk 2]
```diff
@@ -560,6 +560,26 @@ class ScmBase {
...
+     * @param  {String}     [config.scmContext]     The scm context name
+     * @return {Promise}                            Resolves when operation completed without failure
+     */
+    openPr(config) {
+        return validate(config, dataSchema.plugins.scm.openPr)
+            .then(() => this._openPr(config));
+    }
+
+    _openPr() {
+        return Promise.reject(new Error('Not implemented'));
```
Comment (tkyi) [inline index.js:581]: I don't think we should make it reject if it's not implemented
unless we plan to implement for `scm-gitlab` and `scm-bitbucket` as well

File: README.md [hunk 3]
```diff
@@ -473,6 +473,19 @@ The display name of scm context
 #### Expected Response
 1. The display name of scm context
 
+### openPr
+| Parameter        | Type  | Required | Description |
+| :-------------   | :---- | :------- | :-------------|
+| config           | Object | Yes | Configuration Object |
+| config.scmUri    | String | Yes | The scm uri (ex: `github.com:1234:branchName`) |
```
Comment (adong) [inline README.md:480]: I'd like to change this `scmUri` to `checkoutUrl`

File: index.js (hunk 2 above)
Comment (adong) [reply index.js:581]: I'd like to know the rational.

For me, I think `Promise.reject` is okay. If `scm-gitlab` and `scm-bitbucket` is not implemented, we should throw errors.

Comment (jithine) [reply index.js:581]: Either ways caller should handle if pr was not opened, and UI should adjust what is displays accordingly. Not able to open PR should not be an error situation. API response should reflect that.

Comment (tkyi) [reply index.js:584]: Rejecting with error will mean we require all scm plugins to have this method implemented
if you do reject with error, we should bump major version of scm-base

**Comment by adong** [comment]
https://github.com/screwdriver-cd/data-schema/pull/393 is now merged, please take a look on fixing the failed tests.
Thanks for using `checkoutUrl`

File: README.md [hunk 4]
```diff
@@ -473,6 +473,21 @@ The display name of scm context
...
+| Parameter          | Type  | Required | Description |
+| :-------------     | :---- | :------- | :-------------|
+| config             | Object | Yes | Configuration Object |
+| config.checkoutUrl | String | Yes | Checkout url for a repo|
+| config.token       | String | Yes | Access token for scm |
+| config.title       | String | Yes | Pullrequest title   |
+| config.message     | String | Yes | Pullrequest message |
+
+#### Expected Outcome
+The array of scm context name (e.g. [github:github.com, gitlab:my-gitlab])
```
Comment (tkyi) [inline README.md:486]: is this accurate?

Comment (klu909) [reply README.md]: fixed

**Pull request merged**